tests/lib/QueryRendererHelper.test.ts

```diff
@@ -41,7 +40,7 @@ No filters supplied. All tasks will match the query.`;
     });
 
     it('should explain a task with global query active', () => {
-        updateSettings({ globalQuery: 'description includes hello' });
+        GlobalQuery.getInstance().set('description includes hello');
```

Comment:
The only uses of `GlobalQuery.getInstance()` in tests should be in `GlobalQuery.test.ts`, to test the behaviour of `GlobalQuery.getInstance()`. For example, it might test that calling `GlobalQuery.getInstance()` twice gets back the same instance.

Other than that, all tests should use `new GlobalQuery()` - and then setting the query is only changing local data, not global data.

One thing I noticed when trying this approach out - see below... - is that this test was not doing any resetting of the GlobalQuery to tidy up after itself...

Now all the data is local, that isn't needed any more.

```diff
Index: tests/lib/QueryRendererHelper.test.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/lib/QueryRendererHelper.test.ts b/tests/lib/QueryRendererHelper.test.ts
--- a/tests/lib/QueryRendererHelper.test.ts	(revision a697c2273d1d27f3087aec59b8d7b9f04a555cf7)
+++ b/tests/lib/QueryRendererHelper.test.ts	(date 1692127942347)
@@ -22,7 +22,7 @@
 
 No filters supplied. All tasks will match the query.`;
 
-        expect(explainResults(query.source)).toEqual(expectedDisplayText);
+        expect(explainResults(query.source, new GlobalQuery())).toEqual(expectedDisplayText);
     });
 
     it('should explain a task with global filter active', () => {
@@ -36,11 +36,12 @@
 Explanation of this Tasks code block query:
 
 No filters supplied. All tasks will match the query.`;
-        expect(explainResults(query.source)).toEqual(expectedDisplayText);
+        expect(explainResults(query.source, new GlobalQuery())).toEqual(expectedDisplayText);
     });
 
     it('should explain a task with global query active', () => {
-        GlobalQuery.getInstance().set('description includes hello');
+        const globalQuery = new GlobalQuery();
+        globalQuery.set('description includes hello');
 
         const source = '';
         const query = new Query({ source });
@@ -53,11 +54,12 @@
 
 No filters supplied. All tasks will match the query.`;
 
-        expect(explainResults(query.source)).toEqual(expectedDisplayText);
+        expect(explainResults(query.source, globalQuery)).toEqual(expectedDisplayText);
     });
 
     it('should explain a task with global query and global filter active', () => {
-        GlobalQuery.getInstance().set('description includes hello');
+        const globalQuery = new GlobalQuery();
+        globalQuery.set('description includes hello');
         GlobalFilter.set('#task');
 
         const source = '';
@@ -73,11 +75,12 @@
 
 No filters supplied. All tasks will match the query.`;
 
-        expect(explainResults(query.source)).toEqual(expectedDisplayText);
+        expect(explainResults(query.source, globalQuery)).toEqual(expectedDisplayText);
     });
 
     it('should explain a task with global query set but ignored without the global query', () => {
-        GlobalQuery.getInstance().set('description includes hello');
+        const globalQuery = new GlobalQuery();
+        globalQuery.set('description includes hello');
 
         const source = 'ignore global query';
         const query = new Query({ source });
@@ -86,7 +89,7 @@
 
 No filters supplied. All tasks will match the query.`;
 
-        expect(explainResults(query.source)).toEqual(expectedDisplayText);
+        expect(explainResults(query.source, globalQuery)).toEqual(expectedDisplayText);
     });
 });
 
@@ -99,14 +102,18 @@
     it('should be the result of combining the global query and the actual query', () => {
         const querySource = 'description includes world';
         const globalQuerySource = 'description includes hello';
-        GlobalQuery.getInstance().set(globalQuerySource);
-        expect(getQueryForQueryRenderer(querySource).source).toEqual(`${globalQuerySource}\n${querySource}`);
+        const globalQuery = new GlobalQuery();
+        globalQuery.set(globalQuerySource);
+        expect(getQueryForQueryRenderer(querySource, globalQuery).source).toEqual(
+            `${globalQuerySource}\n${querySource}`,
+        );
     });
 
     it('should ignore the global query if "ignore global query" is set', () => {
-        GlobalQuery.getInstance().set('path includes from_global_query');
-        expect(getQueryForQueryRenderer('description includes from_block_query\nignore global query').source).toEqual(
-            'description includes from_block_query\nignore global query',
-        );
+        const globalQuery = new GlobalQuery();
+        globalQuery.set('path includes from_global_query');
+        expect(
+            getQueryForQueryRenderer('description includes from_block_query\nignore global query', globalQuery).source,
+        ).toEqual('description includes from_block_query\nignore global query');
     });
 });
```